modules/consensus/synchronize.go

```diff
@@ -260,3 +273,96 @@ func (cs *ConsensusSet) relayBlock(conn modules.PeerConn) error {
 	}
 	return nil
 }
+
+// rpcRelayHeader is an RPC that accepts a block header from a peer.
+func (cs *ConsensusSet) rpcRelayHeader(conn modules.PeerConn) error {
+	// Decode the block header from the connection.
+	var h types.BlockHeader
+	err := encoding.ReadObject(conn, &h, types.BlockHeaderSize)
+	if err != nil {
+		return err
+	}
+
+	// Start verification inside of a bolt View tx.
+	cs.mu.RLock()
+	err = cs.db.View(func(tx *bolt.Tx) error {
+		// Do not accept a header if the database is inconsistent.
+		if inconsistencyDetected(tx) {
+			return errInconsistentSet
+		}
```

Comment:
I don't think you need this. It's more of a startup sanity check, and was a bigger issue before we were atomically modifying the database.